src/Features/CSharp/Portable/ConvertProgram/ConvertProgramTransform_ProgramMain.cs

```diff
@@ -86,26 +85,35 @@ public static async Task<Document> ConvertToProgramMainAsync(Document document,
             if (method.ParameterList.Parameters.Count == 1 && method.ParameterList.Parameters[0].Type is ArrayTypeSyntax arrayType)
                 method = method.ReplaceNode(arrayType.ElementType, PredefinedType(Token(SyntaxKind.StringKeyword)));
 
-            return generator.ClassDeclaration(
+            return FixupComments(generator.ClassDeclaration(
                 WellKnownMemberNames.TopLevelStatementsEntryPointTypeName,
                 accessibility: accessibilityModifiersRequired ? programType.DeclaredAccessibility : Accessibility.NotApplicable,
                 modifiers: hasExistingPart ? DeclarationModifiers.Partial : DeclarationModifiers.None,
-                members: new[] { method });
+                members: new[] { method }).WithLeadingTrivia(leadingTrivia));
         }
 
-        private static ImmutableArray<StatementSyntax> GenerateProgramMainStatements(CompilationUnitSyntax root)
+        private static ImmutableArray<StatementSyntax> GenerateProgramMainStatements(
+            CompilationUnitSyntax root, out SyntaxTriviaList triviaToMove)
         {
             using var _ = ArrayBuilder<StatementSyntax>.GetInstance(out var statements);
 
+            triviaToMove = default;
             var first = true;
             foreach (var globalStatement in root.Members.OfType<GlobalStatementSyntax>())
             {
-                // Remove leading trivia from first statement.  We'll move it to the Program type.
+                // Remove leading trivia from first statement.  We'll move it to the Program type. Any directly attached
+                // comments though stay attached to the first statement.
                 var statement = globalStatement.Statement.WithAdditionalAnnotations(Formatter.Annotation);
                 if (first)
                 {
                     first = false;
-                    statements.Add(statement.WithoutLeadingTrivia());
+
+                    triviaToMove = statement.GetLeadingTrivia();
+                    while (triviaToMove is [.., { RawKind: (int)SyntaxKind.SingleLineCommentTrivia }, { RawKind: (int)SyntaxKind.EndOfLineTrivia }])
```

Comment:
Are we allowed to use new language features before their GA?

Reply:
Yes.  We can use whatever features are available in the SDK we reference.